modules/tensor_mechanics/doc/content/documentation/modules/tensor_mechanics/generalized_plane_strain.md

```diff
@@ -0,0 +1,78 @@
+# Generalized Plane Strain
+
+## Description
+
+The generalized plane strain problem has found use in many important applications. Bearing the same name as generalized plane strain problem, different presentations have been found in the literature. The simplest one is an extension of the plane strain problem by allowing a non-vanishing constant direction strain in the out-of-plane direction [citep:Adams1967gps]. This has been further generalized by allowing two rotations degrees of freedom [citep:Abaqus2014gps]. An even more generalized form includes the anticlastic problem associated with the out-of-plane shear [citep:Adams1984gps, Li1999gps]. In [tensor mechanics module](tensor_mechanics/index.md), the simplest form was implemented, i.e. an extra degree of freedom representing the out-of-plane direct strain is included in addition to the conventional plane strain problem [citep:Li2005gps].
+
+## Formulation
+
+The deformation of a cylindrical body can be treated as a generalized plane strain problem if it is free from body force and surface traction in the longitudinal direction, while all loads, including body forces and surface traction, are properly balanced in any transverse plane and remain constant along the longitudinal direction. For the description of the problem, it is convenient to introduce the coordinate system so that the $x$-$y$ plane is positioned transverse to the cylindrical body and the $z$-axis runs in the longitudinal direction. Under the conditions described, the in-plane stresses are not functions of $z$. Implied is that the in-plane strains are not functions of $z$ from the constitutive relationship for linearly elastic and homogeneous materials. The generalized plane strain problem is 2D in presentation, defined in a 2D domain $A$ in the $x$-$y$ plane, which is the cross-section of the cylindrical body.
```

Comment:
You don't really need to restrict this discussion to a cylindrical body, do you?